osquery/dispatcher/query_profiler.h

```diff
@@ -0,0 +1,21 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#pragma once
+
+#include <functional>
+#include <string>
+
+#include <osquery/status.h>
+
+namespace osquery {
+void launchQueryWithProfiling(const std::string& name,
```

Comment:
Would you like to pass the return status of `launchQuery` as a return status here?

Reply:
Nice catch